src/Compilers/Core/Portable/SourceGeneration/AdditionalSourcesCollection.cs

```diff
@@ -100,11 +99,11 @@ public bool Contains(string hintName)
 
         internal ImmutableArray<GeneratedSourceText> ToImmutableAndFree() => _sourcesAdded.ToImmutableAndFree();
 
-        private static string AppendExtensionIfRequired(string hintName)
+        private string AppendExtensionIfRequired(string hintName)
         {
-            if (!hintName.EndsWith(".cs", _hintNameComparison))
+            if (!hintName.EndsWith(_fileExtension, _hintNameComparison))
```

Comment:
Based on this it seems like we should assert in the ctor that the provided extension begins with `"."`